improver_tests/developer_tools/test_MOMetadataInterpreter.py

```diff
@@ -0,0 +1,505 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the MOMetadataInterpreter plugin"""
+
+import numpy as np
+import pytest
+from iris.coords import CellMethod
+
+## Test successful outputs (input cubes in alphabetical order by fixture)
+
+
+def test_realizations(ensemble_cube, interpreter):
+    """Test interpretation of temperature realizations from MOGREPS-UK"""
+    interpreter.run(ensemble_cube)
+    assert interpreter.prod_type == "gridded"
+    assert interpreter.field_type == "realizations"
+    assert interpreter.diagnostic == "air_temperature"
+    assert interpreter.relative_to_threshold is None
+    assert not interpreter.methods
+    assert not interpreter.post_processed
+    assert interpreter.model == "MOGREPS-UK"
+    assert not interpreter.blended
+    assert not interpreter.warnings
+
+
+def test_improver_in_title(ensemble_cube, interpreter):
+    """Test that an unblended title including 'IMPROVER' rather than a model name
+    does not cause an error, and that the source model is still identified via the
+    appropriate attribute"""
+    ensemble_cube.attributes["title"].replace("MOGREPS-UK", "IMPROVER")
+    interpreter.run(ensemble_cube)
+    assert interpreter.model == "MOGREPS-UK"
+
+
+def test_static_ancillary(landmask_cube, interpreter):
+    """Test interpretation of static ancillary"""
+    interpreter.run(landmask_cube)
+    assert interpreter.prod_type == "gridded"
+    assert interpreter.field_type == "ancillary"
+    assert interpreter.diagnostic == "land_binary_mask"
+    assert interpreter.relative_to_threshold is None
+    assert not interpreter.methods
+    assert interpreter.post_processed is None
+    assert interpreter.model is None
+    assert not interpreter.blended
+    assert not interpreter.warnings
+
+
+def test_percentiles(wind_gust_percentile_cube, interpreter):
+    """Test interpretation of wind gust percentiles from MOGREPS-UK"""
+    interpreter.run(wind_gust_percentile_cube)
+    assert interpreter.prod_type == "gridded"
+    assert interpreter.field_type == "percentiles"
+    assert interpreter.diagnostic == "wind_gust"
+    assert interpreter.relative_to_threshold is None
+    assert not interpreter.methods
+    assert not interpreter.post_processed
+    assert interpreter.model == "MOGREPS-UK"
+    assert not interpreter.blended
+    assert not interpreter.warnings
+
+
+def test_precip_accum(precip_accum_cube, interpreter):
+    """Test interpretation of nowcast precipitation accumulations"""
+    interpreter.run(precip_accum_cube)
+    assert interpreter.prod_type == "gridded"
+    assert interpreter.field_type == "realizations"
```

Comment:
I'm not sure. Deterministic data are really just a special case of realizations data where there is no uncertainty information available. I think the distinction, if we want to make it, should be that when a realization coord is present (scalar or dim), then we have realizations. Otherwise, it is deterministic. Or we could rename it "diagnostic-space". 
There are some cubes in IMPROVER that are both probabilistic AND have a realization coord (partly-processed precip data for the precip-phase chains). There are no tests covering this case and the interpreter doesn't (as far as I can tell) test the realization coord for such a cube.

Reply:
The interpreter correctly identifies probabilistic data with a realization coordinate as probabilistic - although I can't remember whether there's a unit test for this.  If you want to rename "realization" as "diagnostic", I think that's fine.  There is a case to distinguish probabilities from percentiles and diagnostic information, but there's no reason to distinguish "diagnostic" from "realization".  They are the same thing, and whether or not the source model was an ensemble or not will be clear elsewhere in the metadata (if there is associated model information).

Like @fionaRust, I actually prefer the word "diagnostic" (as in "diagnostic space").  I chose to use the word "realization" because I had found in the past several team members did not understand the phrase "new diagnostic", back when I was introducing the "create new diagnostic cube" functionality to ensure correct metadata.  Hence naming the variable `self.DIAG`, but giving the string output a value "realization" as a translation for users.  But if the two of you think "diagnostic" is better, then please go ahead and replace this string!

Reply:
Thanks Caroline. I agree that I was wrong about the untested cube. Nothing needs adding for the case I described. Fiona and I would like to change the printed output to say "at least one realization", which we think we can do although the grammar may need special handling. This should reduce any future confusion while maintaining the principles you've set out. Do you agree?

Reply:
Sounds fine - you are the users!  Suggest dealing with this in the display function.

Reply:
Now done